server.py

```diff
@@ -50,6 +51,67 @@ def get_shoes(self):
         conn.close()
         return jsonify(data)
     
+    def post_shoes(self):
+        conn = self.__get_db_connection()
+        cursor = conn.cursor()
+
+        #     cursor.execute('''
+        #     CREATE TABLE IF NOT EXISTS shoes (
+        #         shoes_id INTEGER PRIMARY KEY,
+        #         brand TEXT NOT NULL,
+        #         shoe_name TEXT NOT NULL,
+        #         shape TEXT NOT NULL,
+        #         size INT NOT NULL,
+        #         menwear TEXT NOT NULL,
+        #         womanwear TEX NOT NULL,
+        #         price INT NOT NULL,
+        #         manufacturer TEXT NOT NULL      
+        #     )
+        # ''')
+        data = request.get_json()
+        #print(data)
+        brand = data['brand']
+        #print(brand)
+        shoe_name = data['shoe_name']
+        #print(shoe_name)
+        shape = data['shape']
+        #print(shape)
+        size = data['size']
+        #print(size)
+        menwear = data['menwear']
+        #print(menwear)
+        womanwear = data['womanwear']
+        #print(womanwear)
+        price = data['price']
+        #print(price)
+        manufacturer = data['manufacturer']
+        #print(manufacturer)
+
+        checklist = {
+            "brand" : brand,
+            "shoe_name" : shoe_name,
+            "shape" : shape,
+            "size" : size,
+            "menwear" : menwear,
+            "womanwear" : womanwear,
+            "price" : price,
+            "manufacturer" : manufacturer
+        }
+
+        for shoeinfo, shoevalue in checklist.items():
+            if not shoevalue:
+                #print(shoevalue)
+                return jsonify({"error":shoeinfo+" "+"is required"})
+        
+        cursor.execute("INSERT INTO shoes(brand,shoe_name,shape,size,menwear,womanwear,price,manufacturer) VALUES(?,?,?,?,?,?,?,?)", (brand,shoe_name,shape,size,menwear,womanwear,price,manufacturer))
+        conn.commit()
```

Comment:
You should put this code block inside _try-except_. Since, if there is some error in conn.execute or conn.commit method, that error should be caught. And in the _except_ block you should rollback the conn.